### PR TITLE
:recycle: migrate project completely to pyproject.toml

### PR DESCRIPTION
**Pull request by ianardee**
## Description

Migrate to pyproject.toml for all setup stuff.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires a change to the official Guide documentation.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -21,8 +21,13 @@ repos:
             "--ignore=D10,D212,D203,D401"
         ]
 
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.18.2
+    hooks:
+      - id: gitleaks
```
Comment (sebastianMindee) [inline .pre-commit-config.yaml:27]: Very nice :)

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,84 @@
...
+authors = [
+  {name = "Mindee", email = "opensource@mindee.com"},
+]
+dynamic = ["version"]
+description = "Mindee API helper library for Python"
+readme = "README.md"
+license = {file = "LICENSE"}
+classifiers = [
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.7",
```
Comment (sebastianMindee) [inline pyproject.toml:16]: We don't officially support this anymore, I believe

**Review by sebastianMindee — CHANGES_REQUESTED** [review]
Remove mentions of python 3.7 if possible

File: pyproject.toml (hunk 2 above)
Comment (ianardee) [reply pyproject.toml:16]: https://github.com/mindee/mindee-api-python/blob/main/.github/workflows/unit-test.yml#L18

Comment (sebastianMindee) [reply pyproject.toml:16]: Mmmh, what do you say we get rid of that? :grin:

Comment (ianardee) [reply pyproject.toml:16]: can do for next major version, it's a breaking change.

**Pull request merged**